### ISSUE TITLE
MissingMethodException on using LoggingAssert Contains

### ISSUE THREAD

**Issue by Dorus** [alefranz/MELT#37]
I updated my xUnit packages today, and got this exception:
```txt
System.MissingMethodException: Method not found: 'Void Xunit.Sdk.EqualException..ctor(System.Object, System.Object)'.
   at Xunit.LoggingAssert.Contains(IEnumerable`1 expectedValues, IEnumerable`1 actualValues)
   at Xunit.LoggingAssert.Contains(String key, Object value, IEnumerable`1 actualValues)
```
on calling this code:

```C#
var log = _loggerFactory.Sink.LogEntries.First();
LoggingAssert.Contains("fileName", "bla.txt", log.Properties);
```

**Comment by alefranz** [comment]
Thanks for the report, I'll have a look!

**Comment by Dorus** [comment]
I ended up doing the check with FluentAssertions like this:

```C#
    var log = _loggerFactory.Sink.LogEntries.Select(e => e.Properties).Should().BeEquivalentTo(new[] {
      new Dictionary<string, string>
      {
        { "{OriginalFormat}", "File {fileName} is not an xml file." },
        { "fileName", "bla.txt" }
      }
    });
```

**Comment by b-straub** [comment]
Same for me 2.4.2 working , 2.5.0 not. Testing on .NET8.0 latest preview with async tests.

**Comment by bemayr** [comment]
@alefranz Maybe the following can help your investigation a bit:

I just had the same problem while debugging Roslyn Analyzer Tests and found out that between xUnit 2.4.2 and 2.5.0 the public interface of `EqualException changed`. [This diff](https://github.com/xunit/assert.xunit/commit/2fbb2c935469de278f76e0a7237d02d1110decd2#diff-72a16d4ab46d650c22210e8835dddfcea1dfa668bd73febbec816134576eaf77) clearly shows the difference in the constructor, which leads to the *MissingMethodException*.

For MELT, the problem is in those lines:
https://github.com/alefranz/MELT/blob/b4fbc549609d35266b932d6beabbde2f2a47c7f0/src/MELT.Xunit/LoggingAssert.cs#L53-L55

It looks like
```csharp
EqualException.ForMismatchedValues(expected, actual)
```
is the new way to construct such an Exception now.

---
As a comparison the Roslyn Analyzer Testing problem stems from:
https://github.com/dotnet/roslyn-sdk/blob/42ce3944fc4488a7ebfbf1f8d395c66808577d77/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Testing.Verifiers.XUnit/EqualWithMessageException.cs#L12-L16

**Comment by alefranz** [comment]
Released [0.9.0](https://github.com/alefranz/MELT/releases/tag/0.9.0).
Any feedback is appreciated.

Thanks for the investigation @bemayr .
I went the easy route for now and built the base exception, manually crafting the message.
Given the discussion on https://github.com/dotnet/roslyn-sdk/issues/1099, it seems like the best solution is to drop the dependency to the assert package. I have some refactoring in progress, so will consider dropping it as part of that.
